cdap-ui/app/cdap/components/DataPrep/TopPanel/PipelineConfigHelper.js

```diff
@@ -401,6 +401,47 @@ function constructSpannerSource(artifactsList, spannerInfo) {
   };
 }
 
+function constructAdlsSource(artifactsList, adlsInfo) {
+  if (!adlsInfo) {
+    return null;
+  }
+
+  let batchArtifact = find(artifactsList, { name: 'adls-plugins' });
+  if (!batchArtifact) {
+    return T.translate(`${PREFIX}.adls`);
+  }
+
+  batchArtifact.version = '[0.9.2, 9.0.0)';
+  let plugin = objectQuery(adlsInfo, 'values', 0);
+
+  let pluginName = Object.keys(plugin)[0];
+
+  plugin = plugin[pluginName];
```

Comment:
Has this worked before? Right now this is the response I get from wrangler 
```
{
  "message": "Success",
  "count": 1,
  "values": [
    {
      "name": "AzureDataLakeStore",
      "type": "source",
      "properties": {
        "copyHeader": "false",
        "schema": "{\"type\":\"record\",\"name\":\"text\",\"fields\":[{\"name\":\"body\",\"type\":\"string\"}]}",
        "path": "xxx",
        "clientId": "xxx",
        "credentials": "xxx,
        "format": "text",
        "refreshTokenURL": "xxx",
        "referenceName": "xxx"
      }
    }
  ],
  "truncated": "false"
}```

Based on this response the `plugin` will actually store only the name of the plugin after this. Which makes `batchPluginInfo` have all the values as `undefined` except the version. I had to remove the `pluginName` reference entirely as in both line 417 and 419 for this to work for me. Am I missing something? 

With the current change it takes me to the pipline but I cannot open any of the nodes in the pipeline.

Reply:
Removing these lines, not required
because earlier object structure was like 
{
   ...
  "values": [
    {
      "ADLS": {
          .....<properties>....
        }
      }
    }
  ],
  ....
}
now info like this
{
  "message": "Success",
  "count": 1,
  "values": [
    {
     .....<properties>....
      }
    }
  ],
  "truncated": "false"
}

Reply:
I don't think this is correct. What if the wrangler has to send the properties for more than one plugin in the future? The first structure that I mentioned is what should be returned (and consistent with other connections) @albertshau @nitinmotgi Did we know about this? Does it differ between connections (I understand some do a little bit today but just verifying)